source/extensions/filters/http/transformation/inja_transformer.h

```diff
@@ -82,15 +83,24 @@ class Extractor : Logger::Loggable<Logger::Id::filter> {
   absl::string_view extract(Http::StreamFilterCallbacks &callbacks,
                             const Http::RequestOrResponseHeaderMap &header_map,
                             GetBodyFunc &body) const;
-
+  std::string extractDestructive(Http::StreamFilterCallbacks &callbacks,
+                      const Http::RequestOrResponseHeaderMap &header_map,
+                      GetBodyFunc &body) const;
+  ExtractionApi::Mode mode() const { return mode_; }
 private:
   absl::string_view extractValue(Http::StreamFilterCallbacks &callbacks,
                                  absl::string_view value) const;
+  std::string replaceIndividualValue(Http::StreamFilterCallbacks &callbacks,
+                                           absl::string_view value) const;
+  std::string replaceAllValues(Http::StreamFilterCallbacks &callbacks,
+                                     absl::string_view value) const;
 
   const Http::LowerCaseString headername_;
   const bool body_;
   const unsigned int group_;
   const std::regex extract_regex_;
+  const std::optional<std::string> replacement_text_;
```

Comment:
I think this could be a `const std::optional<const std::string&>` since the string is from the config object const ref passed into the constructor

Reply:
Updated here: https://github.com/solo-io/envoy-gloo/pull/309/commits/5fcb54524240eac9fb9f56b609e1aa0d829513f4#diff-7ac442c110145b36211bbfa1c31896337c06b25a47de63177a8943c48b380de8R102

I hit some compiler issues trying to use `const std::optional<const std::string&>`, but `const std::optional<const std::string>` works fine